rasa/core/policies/flow_policy.py

```diff
@@ -149,6 +210,21 @@ def predict_action_probabilities(
              The prediction.
         """
         executor = FlowExecutor.from_tracker(tracker, flows or FlowsList([]))
+        if len(tracker.events) > 0 and tracker.events[-1].type_name == "user":
+            logger.info(flows)
+            flow_prompt = render_template(tracker, flows or FlowsList([]))
+            logger.info(flow_prompt)
+            action_list = generate_text_openai_chat(flow_prompt)
+            logger.info(action_list)
+            prediction = parse_action_list(action_list, flows or FlowsList([]))
```

Comment:
the way this is implemented we can't run it before the flow execution can we? it is either or but ideally we'd extract e.g. slot updates and then continue with the flow advancing